### DISCUSSION TITLE
Contents of Windows docker images

### DISCUSSION THREAD

**Discussion by lanctot** [pypa/cibuildwheel#970] [Q&A] [answered]
Apologies if this is the wrong place to ask.

Is there any place to see what windows images are available and what compilers and versions each has?

I am looking into Windows support for our project, and we use the abseil C++ library which is only officially supported via MSVC. So I am wondering if MSVC is installed on any of them.. if not which ones are? Etc.

Thanks in advance!

**Comment by henryiii** [comment, marked as answer]
Depends on your CI system. What are you planning on using? Pre-installed software for GitHub Actions and Azure runners can be seen here: https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners - you can generally find similar lists with other CI systems.

You really should always be using MSVC, as other compilers are not officially supported (though you generally can make them work, more or less).

cibuildwheel just uses what's on the system; Python is pre-configured to use a recent MSVC; you can manually set up others (such as with https://github.com/ilammy/msvc-dev-cmd for GitHub Actions).

**Reply by lanctot** [reply]
Apologies, I should have included more context: Github Actions. 

We have cibuildwheel working quite well for Linux and MacOS builds, and we have used them for several releases already: https://github.com/deepmind/open_spiel/tree/master/.github/workflows (you help me set them up, actually, thanks again).

It has been a while since I have worked with development in Windows so I was not quite sure about the legalities of having MSVC on these public images (last time I used Visual Studio was in 2003-2004 and IIRC the Microsoft compiler was proprietary and not freely available)

**Reply by henryiii** [reply]
GitHub and Azure are both owned by Microsoft. :)

**Reply by henryiii** [reply]
And there are free and professional versions available.
